src/main/java/org/assertj/assertions/generator/BaseAssertionGenerator.java

```diff
@@ -319,14 +322,49 @@ private String generateAssertionsEntryPointClassContent(final Set<ClassDescripti
         ? determineBestEntryPointsAssertionsClassPackage(classDescriptionSet)
         : entryPointClassPackage;
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, PACKAGE, classPackage);
-
+    Set<String> imports;
+    if (entryPointAssertionsClassContent.contains(IMPORTS)) {
+      imports = extractImports(classDescriptionSet);
+      entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, IMPORTS, listNeededImports(imports));
+    } else {
+      imports = new HashSet<>();
+    }
     String allEntryPointsAssertionContent = generateAssertionEntryPointMethodsFor(classDescriptionSet,
-                                                                                  entryPointAssertionMethodTemplate);
+                                                                                  entryPointAssertionMethodTemplate,
+                                                                                  imports);
     entryPointAssertionsClassContent = replace(entryPointAssertionsClassContent, ALL_ASSERTIONS_ENTRY_POINTS,
                                                allEntryPointsAssertionContent);
     return entryPointAssertionsClassContent;
   }
 
+  private static String listNeededImports(Collection<String> imports) {
```

Comment:
I see that we need the set of imports later in `generateAssertionEntryPointMethodsFor` but I think it is better to recompute them in `generateAssertionEntryPointMethodsFor` as we are passing the `Set<ClassDescription>` which is enough to recompute the imports.
I like when we only pass the minimal parameters to a method, to generate `assertThat` methods we only need the template and the `Set<ClassDescription>`.